XIVComboExpanded/Combos/VPR.cs

```diff
@@ -7,23 +7,340 @@ internal static class VPR
+            DreadMaw = 35,
+            Slither = 40,
+            HuntersBite = 40,
+            SwiftskinsBike = 45,
+            AoE3rdCombo = 50,    // Jagged Maw and Bloodied Maw
+            DeathRattle = 55,
+            LastLash = 60,
+            Dreadwinder = 65,    // Also includes Hunter's Coil and Swiftskin's Coil
+            PitOfDread = 70,     // Also includes Hunter's Den and Swiftskin's Den
+            TwinsSingle = 75,    // Twinfang Bite and Twinblood Bite
+            TwinsAoE = 80,       // Twinfang Thresh and Twinblood Thresh
+            UncoiledFury = 82,
+            UncoiledTwins = 92,  // Uncoiled Twinfang and Uncoiled Twinblood
+            SerpentsIre = 86,
+            EnhancedRattle = 88, // Third stack of Rattling Coil can be accumulated
+            Reawaken = 90,       // Also includes First Generation through Fourth Generation
+            Ouroboros = 96,
+            Legacies = 100;      // First through Fourth Legacy
+    }
+}
+
+internal class SteelTailFeature : CustomCombo
+{
+    protected internal override CustomComboPreset Preset { get; } = CustomComboPreset.ViperSteelTailFeature;
+
+    protected override uint Invoke(uint actionID, uint lastComboMove, float comboTime, byte level)
+    {
+        if (actionID == VPR.SteelFangs)
+        {
+            if (OriginalHook(VPR.SerpentsTail) == VPR.DeathRattle)
+                return VPR.DeathRattle;
+        }
+
+        if (actionID == VPR.DreadFangs)
+        {
+            if (OriginalHook(VPR.SerpentsTail) == VPR.DeathRattle)
+                return VPR.DeathRattle;
+        }
+
+        return actionID;
+    }
+}
+
+internal class SteelTailAoEFeature : CustomCombo
+{
+    protected internal override CustomComboPreset Preset { get; } = CustomComboPreset.ViperSteelTailAoEFeature;
+
+    protected override uint Invoke(uint actionID, uint lastComboMove, float comboTime, byte level)
+    {
+        if (actionID == VPR.SteelMaw)
+        {
+            if (OriginalHook(VPR.SerpentsTail) == VPR.LastLash)
+                return VPR.LastLash;
+        }
+
+        if (actionID == VPR.DreadMaw)
+        {
+            if (OriginalHook(VPR.SerpentsTail) == VPR.LastLash)
+                return VPR.LastLash;
+        }
+
+        return actionID;
+    }
+}
+
+internal class TwinCoilFeature : CustomCombo
+{
+    protected internal override CustomComboPreset Preset { get; } = CustomComboPreset.ViperTwinCoilFeature;
+
+    protected override uint Invoke(uint actionID, uint lastComboMove, float comboTime, byte level)
+    {
+
+        if (actionID == VPR.HuntersCoil)
+        {
+            if (OriginalHook(VPR.Twinfang) == VPR.TwinfangBite)
+                return VPR.TwinfangBite;
+        }
+
+        if (actionID == VPR.SwiftskinsCoil)
+        {
+            if (OriginalHook(VPR.Twinblood) == VPR.TwinbloodBite)
+                return VPR.TwinbloodBite;
+        }
+
+        return actionID;
+    }
+}
```

Comment:
Interesting.  Is the intent for this that you press both of them to do the 1-2 Twin combo?  So if Hunter's Coil is on 1 and Swiftskin's Coil is on 2, from this it would appear that the key sequence after using Dreadwinder would be 1>1>2>2>2>1 (or vice versa), is that the intent?  Given the Twins-into-Serpent combo, I'm a bit surprised this doesn't handle _both_ of the Twin follow-ups on each one.  ie. Hunter's Coil turns into Twinfang Bite and then Twinblood Bite, in sequence, and the reverse for Swiftskin's Coil.  You should be able to identify which one should be next by looking for the Hunter's Venom (Twinfang) and Swiftskin's Venom (Twinblood) status effects.

Reply:
Yeah, the current intent was to turn them into their respective starters, a bit like how you can alternate keys for RPR's Gibbet/Gallows, and it kind of mirrors what VPR does with its base combo alternating 2 buttons.

Yeah, it's probably worthwhile to have both swap in properly on a single button as an option too

Reply:
Maybe even have a function to put those on the same steel fangs/dread fangs buttons so basically consolidating all the followups onto the same button set

Reply:
Ya, once we have the gauge added, my intent was to add a combo that moves them down to Steel and Dread Fangs.  In part because I have on the Shift + number bind above Dread and Steel, and have a _terrible_ habit of accidentally hitting Dread or Steel instead of Hunters or Swiftskins and yeeting my meaty attacks. >.<  It also mirrors how I have Reaper set up, where Gibbet and Gallows are merged down onto my Slice and Soul Slice buttons when available.

Reply:
> Interesting. Is the intent for this that you press both of them to do the 1-2 Twin combo? So if Hunter's Coil is on 1 and Swiftskin's Coil is on 2, from this it would appear that the key sequence after using Dreadwinder would be 1>1>2>2>2>1 (or vice versa), is that the intent? Given the Twins-into-Serpent combo, I'm a bit surprised this doesn't handle _both_ of the Twin follow-ups on each one. ie. Hunter's Coil turns into Twinfang Bite and then Twinblood Bite, in sequence, and the reverse for Swiftskin's Coil. You should be able to identify which one should be next by looking for the Hunter's Venom (Twinfang) and Swiftskin's Venom (Twinblood) status effects.

IMO, either way would be fine, and maybe eventually we should offer the option for people to pick if they want a more interactive or more consolidated version.

Reply:
Ya, definitely, having both on each should be a sub-option, imo. If not here, then in a follow-up.

Reply:
Yeah, it does feel easier to press when you can't mess it up by oGCD clipping, but it does feel more in the "spirit" to have it something you can weave